RecoTauTag/RecoTau/src/PositionAtECalEntranceComputer.cc

```diff
@@ -2,37 +2,137 @@
 
 #include "MagneticField/Engine/interface/MagneticField.h"
 #include "MagneticField/Records/interface/IdealMagneticFieldRecord.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
 #include "CommonTools/BaseParticlePropagator/interface/BaseParticlePropagator.h"
+#include "DataFormats/GeometrySurface/interface/Plane.h"
+#include "TrackPropagation/RungeKutta/interface/defaultRKPropagator.h"
 
 #include <cassert>
 
-PositionAtECalEntranceComputer::PositionAtECalEntranceComputer() : bField_z_(-1.) {}
+//HGCal helper classes
+//MB: looks be copy of HGCal utils: L1Trigger/L1THGCalUtilities/plugins/ntuples/HGCalTriggerNtupleGen.cc
+namespace hgcal_helpers {
+  class SimpleTrackPropagator {
+  public:
+    SimpleTrackPropagator(MagneticField const* f) : field_(f), prod_(field_, alongMomentum, 5.e-5), absz_target_(0) {
+      ROOT::Math::SMatrixIdentity id;
+      AlgebraicSymMatrix55 C(id);
+      C *= 0.001;
```

Comment:
what is the meaning of 0.001 here?

Reply:
As far as I understand it defines uncertainty of starting point used then in trajectory propagation. I mean it is to scale down identity (unity) matrix created in l. 20 by 0.001. Then the scaled matrix is used to define uncertainty of starting point of extrapolation. I think it is an approximation used by an original author and it looks working sensibly. Do you have a proposal how to improve it? There was already a question from @slava77 about possible replacements of this simple propagator recommended by HGCal experts, but w/o feedback (see https://github.com/cms-sw/cmssw/pull/31470#discussion_r493942055).

Reply:
My suggestion is to put this information in a comment, so the meaning is clear in the future.